src/statisticBlock.js

```diff
@@ -97,69 +98,103 @@ class StatisticBlock extends Component {
       }
     }
   }
-
   checkRequiredSize(){
     let element = this.props.element;
-
     let scrollWidth = element.scrollWidth * 0.95;
     let scrollHeight = element.scrollHeight * 0.95;
 
     if(this.props.options.autoSize) {
       let size = this.state.size;
-      let elementClientWidth = this.props.element.clientWidth;
-      let elementClientHeight = this.props.element.clientHeight;
-      let clientWidth = this.state.clientWidth;
-      let clientHeight = this.state.clientHeight;
-      let childHeight = 0;
-
-      if(element.clientHeight == element.scrollHeight
-      && this.state.size == this.props.options.size
-      && !this.state.overflow) return;
-
-      if(this.refs['child-0']) {
-        childHeight = ReactDOM.findDOMNode(this.refs['child-0']).clientHeight;
-      }
+      if(this.props.kpis.qDimensionInfo.length > 0) {
+        let containerElm = this.refs.parent;
+
+        let elementClientWidth = containerElm.getBoundingClientRect().width;
+        let elementClientHeight = containerElm.getBoundingClientRect().height;
```

Comment:
Do `containerElm.getBoundingClientRect()` once as it's expensive, and then read the width and height from that.